Create swift classes protocols and tuples (#424)
Codecademy/docs: content/swift/concepts/classes/classes.md

---
Title: 'Classes'

> Description: 'Classes serve as a template to define characteristics and behaviors of an object.'

Reviewer: ```suggestion
Description: 'Defines characteristics and behaviors of an object.'
```
```suggestion
Description: 'Classes serve as a template to define characteristics and behaviors of an object.'
```

Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Classes'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

Classes serve as a template to define characteristics and behaviors of an object.

They are _reference types_, meaning all _instances_ of a class carry the same characteristics and behaviors.

An instance is an object created based on a class.

## Syntax

In Swift, classes are defined using the `class` keyword, followed by a class name, and a pair of curly braces `{}`.

```swift
class MyClass {
  // Class body
}
```

**Note:** Class names should be capitalized as per convention.

### `var`, `init()`, and `func`

> Variables can be defined using the keyword `var`, followed by the variable name, and a value assignment.

Author: @Dusch4593 Is spacing needed for this ordered list?

> `init()` is a special method that gets called when an instance is created. It can take any number of arguments in the form of `property: DataType`. The `self` keyword refers to the instance itself.

Reviewer: ```suggestion
`init()` is a special method that gets called when a class instance is created. It can take any number of arguments in the form of `property: DataType`. The `self` keyword refers to the instance itself.
```
```suggestion
`init()` is a special method that gets called when an instance is created. It can take any number of arguments in the form of `property: DataType`. The `self` keyword refers to the instance itself.
```

> Functions within a class use the `func` keyword, followed by the function's name, and a set of curly braces `{}` to hold the function's body.

Reviewer: ```suggestion
Functions within a class are defined with the following steps:

1. Use the `func` keyword

2. Name the function.

 3. Provide a set of curly braces `{}` to hold the function's body.
```

```swift
class Employee {
  var name = ""
  var salary = 0
  var manager = false

  init(name: String, salary: Int) {
    self.name = name
    self.salary = salary
  }

  func promote() {
    self.manager = true
  }
}
```

In the code above, `Employee` class is given three properties or variables:

> - `name`: Set to an empty string `""`.
> - `salary`: Set to the integer `0`.
> - `manager`: Set to the boolean value `false`.

Reviewer: ```suggestion
- A `name` set to an empty string `""`.
- A `salary` set to an integer value of `0`.
- A `manager` boolean value set to `false`.
```
```suggestion
- `name`: Set to an empty string `""`.
- `salary`: Set to the integer `0`.
- `manager`: Set to the boolean value `false`.
```

The `init()` method takes in `name` and `salary` arguments that are of data type string and integer, respectively.

> When called on an instance, the `.promote()` function will set its `manager` property to true.

Reviewer: ```suggestion
When called on an instance, the `.promote()` function will set its `manager` property to `true`.
```

## Creating an Instance

In Swift, the `init()` method is called when a class instance is created.

> Sandy Cheeks works for "Tree Dome Enterprises Limited" and makes 10,000 acorns a year. Below, `sandyCheeks` variable is assigned to the value of `Employee()`. This creates an instance of the class `Employee`.

Reviewer: ```suggestion
An employee by the `name` of `"Sandy Cheeks"` works for "Tree Dome Enterprises Limited" and makes a `salary` of 10,000 acorns a year. Below, the `sandyCheeks` variable is assigned to the value of `Employee()`. This creates an instance of the class `Employee`.
```

I would use

Sandy's name and salary are set through the initialization method:

```swift
var sandyCheeks = Employee(name: "Sandy Cheeks", salary: 10000)
```

> **Note:** Classes are also _named types_: types that are given a name when defined and can be referred to as such, i.e. instances of `Employee` have the type `Employee`.

Reviewer: ```suggestion
**Note:** Classes are also _named types_, which are types that are given a name when defined and can be referred to as such (i.e. instances of `Employee` have the type `Employee`).
```
```suggestion
**Note:** Classes are also _named types_: types that are given a name when defined and can be referred to as such, i.e. instances of `Employee` have the type `Employee`.
```

---
Title: 'Classes'

> Description: 'Defines characteristics and behaviors of an object.'

Reviewer 2: For General entries, I think we can start with the noun:

`Description: 'Classes in Swift defines characteristics and behaviors of an object.'`

Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:

>   - 'Classes'

Reviewer 2: also add:

`- 'Objects'`

CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

Classes serve as a template to define characteristics and behaviors of an object.

> They are _reference types_, meaning all _instances_ of a class carry the same characteristics and behaviors.

Reviewer 2: Product team told us to stop using italics for keywords in Docs btw.

An instance is an object created based on a class.

## Syntax

In Swift, classes are defined using the `class` keyword, followed by a class name, and a pair of curly braces `{}`.

```swift
class MyClass {
  // Class body
}
```

**Note:** Class names should be capitalized as per convention.

### `var`, `init()`, and `func`

Variables can be defined with the following steps:

1. Use the `var` keyword.
2. Provide a variable name.
3. Assign a value with `=`.

`init()` is a special method that gets called when a class instance is created. It can take any number of arguments in the form of `property: DataType`. The `self` keyword refers to the instance itself.

Functions within a class are defined with the following steps:

> 1. Use the `func` keyword

Reviewer 2: Add period.

2. Name the function.
3. Provide a set of curly braces `{}` to hold the function's body.

```swift
class Employee {
  var name = ""
  var salary = 0
  var manager = false

  init(name: String, salary: Int) {
    self.name = name
    self.salary = salary
  }

  func promote() {
    self.manager = true
  }
}
```

In the code above, `Employee` class is given three properties or variables:

- A `name` set to an empty string `""`.
- A `salary` set to an integer value of `0`.
- A `manager` boolean value set to `false`.

The `init()` method takes in `name` and `salary` arguments that are of data type string and integer, respectively.

> When called on an instance, the `.promote()` function will set its `manager` property to `true`.

Reviewer 2: function should be method, right?

## Creating an Instance

In Swift, the `init()` method is called when a class instance is created.

> An employee by the `name` of `"Sandy Cheeks"` works for "Tree Dome Enterprises Limited" and makes a `salary` of 10,000 acorns a year. Below, the `sandyCheeks` variable is assigned to the value of `Employee()`. This creates an instance of the class `Employee`.

Reviewer 2: love it

Sandy's name and salary are set through the initialization method:

```swift
var sandyCheeks = Employee(name: "Sandy Cheeks", salary: 10000)
```

> **Note:** Classes are also _named types_, which are types that are given a name when defined and can be referred to as such (i.e. instances of `Employee` have the type `Employee`).

Reviewer 2: take out italics
